src/mui/button/SaveButton.js

```diff
@@ -6,20 +6,23 @@ import ContentSave from 'material-ui/svg-icons/content/save';
 import CircularProgress from 'material-ui/CircularProgress';
 import translate from '../../i18n/translate';
 
-class SaveButton extends Component {
+export class SaveButton extends Component {
 
     handleClick = (e) => {
         if (this.props.saving) {
             // prevent double submission
             e.preventDefault();
+        } else if (!this.props.submitOnEnter && this.props.handleSubmit) {
+            this.props.handleSubmit();
```

Comment:
why not use redux-form's `submit()` action creator instead? That way, you don't have to pass `handleSubmit()` all the way down.